media-sound/mixxx/mixxx-9999.ebuild

```diff
@@ -1,127 +1,180 @@
-	media-libs/portmidi
-	media-libs/rubberband
-	media-libs/taglib:=
-	media-libs/vamp-plugin-sdk
-	sci-libs/fftw:3.0=
-	sys-power/upower
+	<media-libs/taglib-2
+	media-sound/lame
 	virtual/glu
 	virtual/libusb:1
-	virtual/opengl
 	virtual/udev
 	x11-libs/libX11
 	aac? (
 		media-libs/faad2
 		media-libs/libmp4v2
 	)
 	ffmpeg? ( media-video/ffmpeg:= )
-	hid? ( dev-libs/hidapi )
 	keyfinder? ( media-libs/libkeyfinder )
 	lv2? ( media-libs/lilv )
-	mp3? ( media-libs/libmad )
-	mp4? ( media-libs/libmp4v2:= )
-	opus? (	media-libs/opusfile )
-	qtkeychain? ( dev-libs/qtkeychain:=[qt5(+)] )
-	shout? ( >=media-libs/libshout-2.4.5 )
+	midi? ( media-libs/portmidi )
+	modplug? ( media-libs/libmodplug )
+	mp3? (
+		media-libs/libid3tag:=
+		media-libs/libmad
+	)
+	mp4? ( media-libs/libmp4v2 )
+	opus? (
+		media-libs/opus
+		media-libs/opusfile
+	)
+	qtkeychain? ( dev-libs/qtkeychain:=[qt6(-)] )
+	rubberband? ( media-libs/rubberband:= )
+	shout? ( dev-libs/openssl:= )
+	upower? (
+		dev-libs/glib:2
+		sys-power/upower:=
+	)
 	wavpack? ( media-sound/wavpack )
 "
 DEPEND="${RDEPEND}
 	dev-cpp/ms-gsl
+	test? ( dev-cpp/gtest )
 "
 BDEPEND="
-	dev-qt/qttest:5
-	dev-qt/qtxmlpatterns:5
+	dev-util/spirv-tools
 	virtual/pkgconfig
 "
+# Not todo to prevent links
+#benchmark? (
+#	dev-cpp/benchmark
+#	dev-util/google-perftools
+#)
 
 PATCHES=(
-	"${FILESDIR}"/${PN}-9999-docs.patch
+	# Building mixxx-test target only with explicit test useflag
+	"${FILESDIR}"/${P}-tests.patch
+	# Fix strict-aliasing violations in vendored katai_cpp_stl_runtime
+	# https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/commit/c01f530.patch
+	"${FILESDIR}"/${PN}-2.5.0-fix-strict-aliasing-kaitai.patch
 )
 
+CMAKE_SKIP_TESTS=(
+	# need HID controller
+	LegacyControllerMappingValidationTest.HidMappingsValid
+)
+
+PLOCALES="
+	ca cs de en es fi fr gl id it ja nl pl pt ro ru sl sq sr tr zh-CN zh-TW
+"
+
+mixxx_set_globals() {
+	local lang
+	local MANUAL_URI_BASE="https://downloads.mixxx.org/manual/${MY_PV}"
+	for lang in ${PLOCALES} ; do
+		SRC_URI+=" l10n_${lang}? ( ${MANUAL_URI_BASE}/${PN}-manual-${MY_PV}-${lang/-/_}.pdf )"
+		IUSE+=" l10n_${lang/ en/ +en}"
+	done
+	SRC_URI+=" ${MANUAL_URI_BASE}/${PN}-manual-${MY_PV}-en.pdf"
+}
+mixxx_set_globals
+
 src_configure() {
+	# prevent ld error as package builds static libs.
+	tc-is-lto && append-flags $(test-flags -ffat-lto-objects)
+
 	local mycmakeargs=(
-		-DFAAD="$(usex aac on off)"
-		-DFFMPEG="$(usex ffmpeg on off)"
-		-DHID="$(usex hid on off)"
-		-DLILV="$(usex lv2 on off)"
-		-DMAD="$(usex mp3 on off)"
-		-DOPTIMIZE="off"
-		-DCCACHE_SUPPORT="off"
-		-DOPUS="$(usex opus on off)"
-		-DBROADCAST="$(usex shout on off)"
-		-DVINYLCONTROL="on"
-		-DINSTALL_USER_UDEV_RULES=OFF
-		-DWAVPACK="$(usex wavpack on off)"
-		-DQTKEYCHAIN="$(usex qtkeychain on off)"
-		-DKEYFINDER="$(usex keyfinder on off)"
-		-DDOWNLOAD_MANUAL=OFF
+		-DBATTERY="$(usex upower ON OFF)"
+		-DBROADCAST="$(usex shout ON OFF)"
 		-DBUILD_SHARED_LIBS=OFF
```

Comment:
I realise that was there before but it is exactly the opposite of what we want. Looking at 9c4c56577c6a096227449c6767c10a97d8a9be9e its author was not sure themselves, and worse, it was to fix a build issue present only in irrelevant benchmarking code. @thesamesam?

Reply:
(only looked at 9999 ebuild yet, not to duplicate efforts)

Reply:
shared_libs is only for live ebuild because there's a lot of work with new components and seems that's not always ready. 2.5.0 is SHARED_LIBS=ON.
But I think 9999 is really a pain for this package.

Reply:
It is generally accepted for 9999 to be intermittently broken by upstream changes, `-DBUILD_SHARED_LIBS=OFF` shouldn't make any difference. If upstream really do not even try to keep git master buildable, then yes we should think about dropping it. We don't benefit from a 9999 ebuild if it can't build on the same options as release versions.

Reply:
As upstream releases often, I don't see any benefit to 9999.

Reply:
+1 for mixxx-9999.
It is true,  upstream releases more often than a few years before and now we have with 2.5.0 the first release meant to be used with QT6.
But there are people who like to live on the edge and try the latest features.
As for all ebuilds it is important to get patches up-streamed to have less breakage.
(@PPN-SD thank you for the upstream PR.)
Users of live ebuilds mostly know how to workaround and report necessary changes to Gentoo or upstream.
The live ebuild is also important to check if an issue in release is already fixed upstream and confirm it works.
I am using my mixxx-9999 since years and upstream was always good in keeping git main build-able.
Most of the changes are merged first into the latest release branch anyway and then after a while into main.